### PR TITLE
Update the pipeline tutorial to include `gradio.Interface.from_pipeline`

### PR DESCRIPTION
**Pull request by abidlabs**
# What does this PR do?

Adds `gradio.Interface.from_pipeline` to the pipeline docs, as [discussed internally](https://huggingface.slack.com/archives/C01N44FJDHT/p1702977938149529?thread_ts=1702943692.389539&cid=C01N44FJDHT).

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: docs/source/en/pipeline_tutorial.md [hunk 1]
```diff
@@ -314,4 +314,28 @@ pipe = pipeline(model="facebook/opt-1.3b", device_map="auto", model_kwargs={"loa
...
 
-Note that you can replace the checkpoint with any of the Hugging Face model that supports large model loading such as BLOOM!
+Note that you can replace the checkpoint with any Hugging Face model that supports large model loading, such as BLOOM.
+
+## Creating web demos from pipelines with `gradio`
+
+Pipelines are automatically supported in [the gradio library](https://github.com/gradio-app/gradio/), making it a breeze to create a web demo
+based on your pipeline. First, make sure you have installed `gradio` with: `pip install gradio`.
+
+Then, create a web demo around an image classification pipeline (or any other pipeline) in a single line of code:
```
Comment (stevhliu) [inline docs/source/en/pipeline_tutorial.md:324]:
```suggestion
Then, you can create a web demo around an image classification pipeline (or any other pipeline) in a single line of code by calling Gradio's [`Interface.from_pipeline`](https://www.gradio.app/docs/interface#interface-from-pipeline) class to launch the pipeline. This creates an intuitive drag-and-drop interface in your browser:
```

File: docs/source/en/pipeline_tutorial.md [hunk 2]
```diff
@@ -314,4 +314,28 @@ pipe = pipeline(model="facebook/opt-1.3b", device_map="auto", model_kwargs={"loa
...
+```py
+from transformers import pipeline
+import gradio as gr
+
+pipe = pipeline("image-classification", model="google/vit-base-patch16-224")
+
+gr.Interface.from_pipeline(pipe).launch()
+```
+
+which opens an intuitive drag-and-drop interface in your browser:
```
Comment (stevhliu) [inline docs/source/en/pipeline_tutorial.md:335]:
```suggestion
```

File: docs/source/en/pipeline_tutorial.md [hunk 3]
```diff
@@ -314,4 +314,28 @@ pipe = pipeline(model="facebook/opt-1.3b", device_map="auto", model_kwargs={"loa
...
+
+![](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/panda-classification.png)
+
+By default, the web demo runs on a local server. If you'd like to share it with others, you can generate a temporary public
+link by setting `share=True` in `launch()`. Or, you can host your demo on [Hugging Face Spaces](https://huggingface.co/spaces)https://huggingface.co/spaces for a permanent link.
```
Comment (stevhliu) [inline docs/source/en/pipeline_tutorial.md:340]:
```suggestion
link by setting `share=True` in `launch()`. You can also host your demo on [Hugging Face Spaces](https://huggingface.co/spaces) for a permanent link. 
```

**Review by stevhliu — APPROVED** [review]
Super cool feature, thanks! 👏

**Comment by abidlabs** [comment]
Thanks @stevhliu for the fast review! Accepted your suggestions and made a couple of small tweaks

File: docs/source/en/pipeline_tutorial.md [hunk 4]
```diff
@@ -314,4 +314,30 @@ pipe = pipeline(model="facebook/opt-1.3b", device_map="auto", model_kwargs={"loa
...
 
-Note that you can replace the checkpoint with any of the Hugging Face model that supports large model loading such as BLOOM!
+Note that you can replace the checkpoint with any Hugging Face model that supports large model loading, such as BLOOM.
+
+## Creating web demos from pipelines with `gradio`
+
+Pipelines are automatically supported in [Gradio](https://github.com/gradio-app/gradio/), a library that makes creating beautiful and user-friendly machine learning apps on the web a breeze. First, make sure you have Gradio installed:
+
+```
+ pip install gradio
```
Comment (amyeroberts) [inline docs/source/en/pipeline_tutorial.md:324]: nit

```suggestion
pip install gradio
```

**Review by amyeroberts — APPROVED** [review]
Thanks for updating!

**Comment by abidlabs** [comment]
Thanks @amyeroberts and @stevhliu ❤️

**Pull request merged**